packages/settingeditor/src/settingspanel.tsx

```diff
@@ -79,6 +93,18 @@ export const SettingsPanel: React.FC<ISettingsPanelProps> = ({
     [id: string]: boolean;
   }> = React.useRef({});
 
+  React.useEffect(() => {
+    // When filter updates, only show plugins that match search.
+    updateFilterSignal.connect(
+      (
+        list: PluginList,
+        newFilter: (plugin: ISettingRegistry.IPlugin) => string[] | null
+      ) => {
+        setFilter(() => newFilter);
+      }
+    );
```

Comment:
You can move it in the below `useEffect`. And please set up a disconnect in the returned callback (see example below).

Reply:
Good point! Done